src/gleam_json_ffi.mjs

```diff
@@ -0,0 +1,70 @@
+import { Ok, Error } from './gleam.mjs'
+import { bit_string_to_string } from '../../gleam_stdlib/dist/gleam_stdlib.mjs'
+import { UnexpectedByte, UnexpectedEndOfInput } from './gleam/json.mjs'
+
+export function json_to_string(json) {
+  return JSON.stringify(json)
+}
+
+export function object_from(entries) {
+  return Object.fromEntries(entries)
+}
+
+export function array(list) {
+  return list.toArray()
+}
+
+export function do_null() {
+  return null
+}
+
+export function string(x) {
+  return json_to_string(x)
+}
+
+export function int(x) {
+  return parseInt(json_to_string(x), 10)
+}
+
+export function float(x) {
+  return parseFloat(json_to_string(x), 10)
+}
+
+export function bool(x) {
+  return Boolean(x)
+}
+
+
+export function decode(bit_string) {
+  const stringResult = bit_string_to_string(bit_string)
+  if (!stringResult.isOk()) return stringResult
+  try {
+    const result = JSON.parse(stringResult[0])
+    return new Ok(result)
+  } catch (err) {
+    return new Error(getJsonDecodeError(err))
+  }
+}
+
+function getJsonDecodeError(stdErr) {
+  if (isUnexpectedByte(stdErr)) return new toUnexpectedByteError(stdErr)
+  if (isUnexpectedEndOfInput(stdErr)) return new UnexpectedEndOfInput()
+  return undefined
+}
+
+function isUnexpectedEndOfInput(err) {
+  return err.message === 'Unexpected end of JSON input'
+}
+
+const unexpectedByteRegex = /Unexpected token (.) in JSON at position (\d+)/
```

Comment:
Is this going to work on all JavaScript runtimes?

Reply:
I think it probably will not. I will need to do more research to figure out how to handle specific runtime errors in JS. This is just a very rough go to see if I could get something working.